src/hooks/usePrefersColorScheme.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react'
+
+type ColorSchemeType = 'dark' | 'light' | 'no-preference'
+
+const usePrefersColorScheme = (): ColorSchemeType => {
+  const [preferredColorScheme, setPreferredColorScheme] = useState<ColorSchemeType>('no-preference')
+
+  useEffect(() => {
+    if (typeof window.matchMedia !== 'function') return
+
+    const isDark = window.matchMedia('(prefers-color-scheme: dark)')
+    const isLight = window.matchMedia('(prefers-color-scheme: light)')
```

Comment:
Aren't they mutually exclusive?

Reply:
From my understanding of https://w3c.github.io/csswg-drafts/mediaqueries-5/#prefers-color-scheme they are not mutually exclusive but correct me if I'm wrong. The user can prefer `dark` or `light` or have no preference.

Reply:
Yeah, but you don't have a third icon for `!isDark && !isLight`, do you?
Also from the docs you linked (emphasis mine):
> `light`
> Indicates that user has expressed the preference for a page that has a light theme (dark text on light background), **or has not expressed an active preference (and thus should receive the "web default" of a light theme)**.